src/Multicall3.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.12;
+
+/// @title Multicall3
+/// @notice Aggregate results from multiple read-only function calls
+/// @dev Multicall & Multicall2 backwards-compatible
+/// @author Michael Elliot <mike@makerdao.com>
+/// @author Joshua Levine <joshua@makerdao.com>
+/// @author Nick Johnson <arachnid@notdot.net>
+/// @author Andreas Bigger <andreas@nascent.xyz>
+
+contract Multicall3 {
+    struct Call {
+        address target;
+        bytes callData;
+    }
+
+    struct Call3 {
+        address target;
+        bytes callData;
+        bool requireSuccess;
+    }
+
+    struct Result {
+        bool success;
+        bytes returnData;
+    }
+
+    function aggregate(Call[] memory calls) public returns (uint256 blockNumber, bytes[] memory returnData) {
+        blockNumber = block.number;
+        uint256 length = calls.length;
+        returnData = new bytes[](length);
+        for (uint256 i = 0; i < length;) {
+            (bool success, bytes memory ret) = calls[i].target.call(calls[i].callData);
+            require(success, "Multicall aggregate: call failed");
+            returnData[i] = ret;
+            unchecked { i++; }
```

Comment:
is ++i faster?

Reply:
<img width="299" alt="Screen Shot 2022-02-28 at 9 18 30 PM" src="https://user-images.githubusercontent.com/21288394/156109645-04293078-cff0-4368-90a6-e64d7019f7bd.png">

I'm frazzled that this saves a gas

Reply:
xd

Reply:
Maybe some people will be interested in why this is like it is: well here we go:
i++ -> Load, Store, Add, Store
++i -> Load, Add, Store